src/proto/tagpost.proto

```diff
@@ -6,18 +6,71 @@ option java_multiple_files = true;
 option java_package = "com.google.tagpost";
 option java_outer_classname = "TagPostProto";
 
+import "google/protobuf/timestamp.proto";
+
 service TagpostService {
+  // dummy rpc service
   rpc FetchMessage(FetchMessageRequest) returns (FetchMessageResponse) {}
+  // fetch all threads by primary tag
+  rpc FetchThreadsByTag(FetchThreadsByTagRequest) returns (FetchThreadsByTagResponse) {}
+  // fetch all comments under one thread
+  rpc FetchCommentsUnderThread(FetchCommentsUnderThreadRequest) returns (FetchCommentsUnderThreadResponse) {}
+  // fetch all comments by a set of tags
+  rpc FetchCommentsByTags(FetchCommentsByTagsRequest) returns (FetchCommentsByTagsResponse) {}
+}
+
+message Tag {
+  string tag_name = 1;
+}
+
+message TagpostThread {
```

Comment:
You can name it Thread directly. It is in package tagpost, so it will not be conflict with other thread class.